components/payments.tsx

```diff
@@ -1,58 +1,137 @@
-import { Badge, Flex, Box, Card, IconButton, Paragraph, Link } from '@theme-ui/components';
+import type { Stripe_PaymentIntent, Stripe_Invoice, Stripe_Item, Stripe_LineItem } from 'lib/takeshape/types';
+import NextLink from 'next/link';
+import { Badge, Flex, Box, Card, IconButton, Paragraph, Link, Heading, Text, Grid } from '@theme-ui/components';
 import { formatPrice } from 'lib/utils/text';
 import { format } from 'date-fns';
 import startCase from 'lodash/startCase';
 import { BsCheckCircleFill, BsReceipt } from 'react-icons/bs';
-
+import { ProductImage } from './product';
 export interface OrderStatusProps {
-  status: 'processing' | 'shipped' | 'delivered';
+  status: 'unknown' | 'in_transit' | 'delivered' | 'error';
   trackingNumber?: string | null;
 }
 
-export const OrderStatus: React.FC<OrderStatusProps> = ({ status, trackingNumber }) => {
-  const colorMap = {
-    processing: 'primary',
-    shipped: 'gray',
-    delivered: 'purple'
+export const OrderStatus = ({ status, trackingNumber }: OrderStatusProps) => {
+  const statusMap = {
+    unknown: {
+      text: 'Processing',
+      color: 'gray'
+    },
+    in_transit: {
+      text: 'Shipped',
+      color: 'green'
+    },
+    delivered: {
+      text: 'Delivered',
+      color: 'purple'
+    },
+    error: {
+      text: 'Error',
+      color: 'red'
+    }
   };
+
+  const statusObj = statusMap[status];
+
   return (
     <Flex sx={{ gap: '.5rem' }}>
-      <Badge py={1} px={2} backgroundColor={colorMap[status] ?? 'gray'}>
-        {startCase(status)}
+      <Badge py={1} px={2} backgroundColor={statusObj.color}>
+        {startCase(statusObj.text)}
       </Badge>
       {trackingNumber && (
-        <Badge py={1} px={2} backgroundColor="gray">
+        <Badge py={1} px={2} backgroundColor={statusObj.color}>
           {trackingNumber}
         </Badge>
       )}
     </Flex>
   );
 };
 
-const ordersFixture: OrderStatusProps[] = [
-  {
-    status: 'processing',
-    trackingNumber: null
-  },
-  {
-    status: 'shipped',
-    trackingNumber: 'Z0123456789R1337'
-  },
-  {
-    status: 'delivered'
-  }
-];
+export interface ProductLineItemProps {
+  id: string;
+  name: string;
+  description: string;
+  images?: string[];
+  amount: number;
+  currency: string;
+  quantity: number;
+}
+
+const ProductLineItem = ({ id, name, description, images, quantity, amount, currency }: ProductLineItemProps) => {
+  return (
+    <Card sx={{ height: '100%', cursor: 'pointer', p: 0 }}>
+      <NextLink href={`/product/${id}`} passHref>
+        <Grid gap={2} columns={['1fr 2fr 1fr']}>
+          <Box sx={{ textAlign: 'left' }}>
+            <Link sx={{ display: 'inline-block' }}>
+              <ProductImage maxHeight="60px" images={images} />
```

Comment:
```suggestion
              <ProductImage maxHeight="30px" images={images} />
```

Smaller image thumbnail to increase vertical density